#3 Impossible to use pin

JasonPUEL opened the pull request:
Link to : Impossible to use pin 26, 27 #3 ticket

JasonPUEL committed:
Fix Impossible to use pin 26, 27 #3

JasonPUEL committed:
Update Log Level to TRACE + small test fix

JasonPUEL committed:
Update Documentation

JasonPUEL committed:
Create Python Non-regretion Test suite

JasonPUEL committed:
Fix test

JasonPUEL committed:
Update Doc on Python Non-Reg test suite

JasonPUEL committed:
Update pyhton test API

JasonPUEL committed:
Update Test of non regression

JasonPUEL committed:
fix keyword 'output error must be FAILURE'

JasonPUEL committed:
Retrieve ticket name from first line description

JasonPUEL committed:
revert last commit

JasonPUEL committed:
Amend

JasonPUEL committed:
Fix requirements: sliplib V0.7.0 not work on this project

JasonPUEL committed:
Use Pytest for python test

soutade commented on firmware/src/main.rs:
```diff
@@ -220,8 +220,8 @@ unsafe fn main() -> ! {
     // --------------------------------------------------------------
-
-    let mut pins_id: [Option<DynPinId>; 23] = [
+    const MAX_PINS: usize = 29;
```
Ok for update, but MAX_PINS const is duplicated

soutade commented on test/robot/libraries/API_PicoHostAdapterDio.py:
```diff
@@ -111,35 +116,58 @@ def __picoha_dio_answer(self):
         """Check if the serial port is open"""
         return self.__serialPort.is_open
 
     def ping_info(self):
```
You can add return type

soutade commented on test/robot/libraries/API_PicoHostAdapterDio.py:
```diff
@@ -111,35 +116,58 @@ def __picoha_dio_answer(self):
 
-    def get_gpio_direction(self, gpio: int) -> int:
-        """Get direction of gpio in INPUT/OUTPUT"""
+    def get_gpio_direction(self, gpio: int):
```
Missing return type

soutade commented on test/robot/libraries/launcher_Python_test.py:
```diff
@@ -0,0 +1,175 @@
+# ================== Variables =================
+
+_results = []
+_DUT = "COM5"
```
Can be set as a parameter

soutade commented on test/robot/libraries/launcher_Python_test.py:
```diff
@@ -0,0 +1,175 @@
+
+def timeout_wrapper(timeout=60):
+    """
+    Launch a fonction and break it on timeout.
```
Typo fonction -> function

soutade commented on test/robot/platform/RaspberryPico/platform.resource:
```diff
@@ -12,13 +12,13 @@ Library     platform_data.py
 
 Main Test Platform Setup
     [Documentation]    Prepare testing environement
-    Set Log Level    DEBUG
```
Not sure is good to have TRACE level

JasonPUEL replied:
This means we save the TRACE level. In the .html file generated by Robot, you can choose which level you want to display. It's really useful for analysis.

soutade commented on test/robot/tests/Test_Suite_NonReg_Git_Issues.py:
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+"""
+Test Suite of no regretion
```
Typo: regretion

soutade commented on test/robot/tests/Test_Suite_NonReg_Git_Issues.py:
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+"""
+Test Suite of no regretion
+Describe small test senarios to help validation of Git issues 
```
Typo senarios

soutade commented on test/robot/tests/Test_Suite_NonReg_Git_Issues.py:
```diff
@@ -0,0 +1,131 @@
+
+# local imports
+# TODO : Remove this ASAP
+sys.path.append(os.path.join(os.path.dirname(os.path.dirname(__file__)), "libraries"))
```
`import .libraries` should works

JasonPUEL replied:
For some reason, that doesn’t work in my venv.

JasonPUEL replied:
setting.json -> "python.analysis.extraPaths" shoulds avoid that problem but don't work for me

soutade commented on test/robot/tests/Test_Suite_NonReg_Git_Issues.py:
```diff
@@ -0,0 +1,131 @@
+from API_PicoHostAdapterDio import PicoHostAdapterDio
+
+# ========= Test Suite of no regretion =========
+# Check Git Issu
```
Typo Issu

soutade commented on test/robot/tests/Test_Suite_NonReg_Git_Issues.py:
```diff
@@ -0,0 +1,131 @@
+    test.ping_info()
+
+    test.set_gpio_direction(gpio=2, direction=dio.PinValue.OUTPUT)
+    test.get_gpio_direction(gpio=2)
```
No assert ?

JasonPUEL committed:
Return from code review :
- Fix MAX_PINS const is duplicated
- Add Return type in API
- Use port com as parameter in test
- Fix tipos

JasonPUEL committed:
Move config file

JasonPUEL committed:
Add extension recommendation file

JasonPUEL committed:
Fix typo

JasonPUEL committed:
Add output msg on assert

JasonPUEL committed:
Update Python

JasonPUEL committed:
Update Robot + Add Test USB COM

JasonPUEL committed:
Add Workspace File

JasonPUEL committed:
Small Cleaning
